src/python/gem5/components/cachehierarchies/abstract_cache_hierarchy.py

```diff
@@ -28,17 +40,72 @@
     ABCMeta,
     abstractmethod,
 )
+from typing import Callable
 
 from m5.objects import SubSystem
+from m5.util.fdthelper import *
 
 from ..boards.abstract_board import AbstractBoard
 
 
+class CacheNode:
+    def __init__(
+        self,
+        name: str,
+        cache: SimObject,
+        next_level: int,
```

Comment:
Shouldn't this be : 
```suggestion
        next_level: "CacheNode",
```

Down in the `add_child`, `self` is passed.